libcontainer/factory_linux.go

```diff
@@ -336,33 +336,33 @@ func (l *LinuxFactory) Type() string {
 // This is a low level implementation detail of the reexec and should not be consumed externally
 func (l *LinuxFactory) StartInitialization() (err error) {
 	var (
-		pipefd, fifofd int
-		consoleSocket  *os.File
-		envInitPipe    = os.Getenv("_LIBCONTAINER_INITPIPE")
-		envFifoFd      = os.Getenv("_LIBCONTAINER_FIFOFD")
-		envConsole     = os.Getenv("_LIBCONTAINER_CONSOLE")
+		it                  initType
+		fifofd              int
+		pipe, consoleSocket *os.File
+
+		envInitPipe = os.Getenv("_LIBCONTAINER_INITPIPE")
+		envInitType = os.Getenv("_LIBCONTAINER_INITTYPE")
+		envFifoFd   = os.Getenv("_LIBCONTAINER_FIFOFD")
+		envConsole  = os.Getenv("_LIBCONTAINER_CONSOLE")
 	)
 
-	// Get the INITPIPE.
-	pipefd, err = strconv.Atoi(envInitPipe)
-	if err != nil {
-		return fmt.Errorf("unable to convert _LIBCONTAINER_INITPIPE=%s to int: %s", envInitPipe, err)
-	}
-
-	var (
-		pipe = os.NewFile(uintptr(pipefd), "pipe")
-		it   = initType(os.Getenv("_LIBCONTAINER_INITTYPE"))
-	)
-	defer pipe.Close()
-
 	// Only init processes have FIFOFD.
 	fifofd = -1
+	it = initType(envInitType)
```

Comment:
Perhaps remove them from the `var(  )` block and just;

```suggestion
	fifofd := -1
	it := initType(envInitType)
```